packages/boa/esm/loader.mjs

```diff
@@ -0,0 +1,39 @@
+import Boa from '../lib/index.js';
+
+const extensionsPrefix = 'py:';
+const passBaseUrl = 'file://py/';
+
+export function resolve(specifier, context, defaultResolve) {
+  if (specifier.startsWith(extensionsPrefix)) {
+    return {
+      url: passBaseUrl + specifier.replace(extensionsPrefix, '')
+    };
+  }
+  return defaultResolve(specifier, context, defaultResolve);
+}
+
+export function getFormat(url) {
+  let format;
+  if (url.startsWith('nodejs:')) {
+    format = 'builtin';
+  } else if (url.startsWith(passBaseUrl)) {
+    format = 'dynamic';
+  }
+
+  return {
+    format: format || 'module'
+  };
+}
+
+export function dynamicInstantiate(url) {
+  const moduleInstance = Boa.import(url.replace(passBaseUrl, ''));
+  const moduleExports = ['default', ...Object.keys(moduleInstance)];
```

Comment:
Missing space for '[]' :)

Reply:
fixed. Maybe we can add some eslint rules.